### PR TITLE
Handle signed[unsigned] conversion for opcodes[GT, SGT, LT, SLT, MOD,…

### PR DESCRIPTION
**Pull request by ngotungson**
… SMOD]

### CODE REVIEW

**Comment by loiluu** [comment]
@inian can you please review this?

File: symExec.py [hunk 1]
```diff
@@ -734,8 +734,9 @@ def sym_exec_ins(start, instr, stack, mem, global_state, path_conditions_and_var
             raise ValueError('STACK underflow')
     elif instr_parts[0] == "MOD":
         if len(stack) > 1:
-            first = stack.pop(0)
-            second = stack.pop(0)
+            first = unsigned_format(stack.pop(0))
```
Comment (loiluu) [inline symExec.py]: have you tested this unsigned_format() and signed_format() functions with symbolic variables?

Comment (inian) [inline symExec.py]: This just seems to be a user defined function and all the operations used seem to be supported by z3 too

File: symExec.py [hunk 2]
```diff
@@ -621,7 +621,7 @@ def sym_exec_ins(start, instr, stack, mem, global_state, path_conditions_and_var
                 first = BitVecVal(first, 256)
             elif not isinstance(first, (int, long)) and isinstance(second, (int, long)):
                 second = BitVecVal(second, 256)
-            computed = first * second & UNSIGNED_BOUND_NUMBER
+            computed = first * second & (2**256 - 1)
```
Comment (inian) [inline symExec.py]: Change this back to the constant?

File: symExec.py (hunk 1 above)
Comment (loiluu) [inline symExec.py]: but if `first` is a symbolic variable, does it work?

File: symExec.py (hunk 2 above)
Comment (ngotungson) [inline symExec.py]: Ok

File: symExec.py (hunk 1 above)
Comment (inian) [inline symExec.py]: Oh yes, the `if` condition needs to changed to handle symbolic variables.

Comment (inian) [inline symExec.py]: @ngotungson check if `first` is symbolic and handle it similar to [this](https://github.com/melonproject/oyente/blob/a333ea2c74c856407e8369fd9e0e3deed14cd128/symExec.py#L901)

**Comment by inian** [comment]
I have run the arithmetic test cases and they seemed to have fixed the cases which were failing before.

**Comment by loiluu** [comment]
yeah it works for the non-symbolic cases.

**Pull request merged**